filewatcher/v2/filewatcher_test.go

```diff
@@ -0,0 +1,612 @@
+		}
+		return value, nil
+	}
+
+	dir := t.TempDir()
+	path := filepath.Join(dir, "foo")
+
+	writeFile(t, path, nil)
+
+	// Initial call to parser should return 1, nil
+	data, err := filewatcher.New(
+		context.Background(),
+		path,
+		parser,
+		filewatcher.WithPollingInterval(-1), // disable polling as we need exact numbers of parser calls in this test
+		filewatcher.WithFSEventsDelay(fsEventsDelayForTests),
+	)
+	if err != nil {
+		t.Fatal(err)
+	}
+	t.Cleanup(func() {
+		data.Close()
+	})
+	expected := int64(1)
+	if value := data.Get(); value != expected {
+		t.Errorf("data.Get() expected %d, got %d", expected, value)
+	}
+
+	// Next call to parser should return nil, err
+	newpath := path + ".bar"
+	writeFile(t, newpath, nil)
+	if err := os.Rename(newpath, path); err != nil {
+		t.Fatal(err)
+	}
+	// Give it some time to handle the file content change
+	time.Sleep(500 * time.Millisecond)
+	if counter.count.Load() == 0 {
+		t.Error("Expected logger being called")
+	}
+	if value := data.Get(); value != expected {
+		t.Errorf("data.Get() expected %d, got %d", expected, value)
+	}
+
+	// Next call to parser should return 3, nil
+	writeFile(t, newpath, nil)
+	if err := os.Rename(newpath, path); err != nil {
+		t.Fatal(err)
+	}
+	// Give it some time to handle the file content change
+	time.Sleep(500 * time.Millisecond)
+	if got, want := data.Get(), int64(3); got != want {
+		t.Errorf("data.Get() got %d, want %d", got, want)
+	}
+}
+
+func updateDirWithContents(tb testing.TB, dst string, contents map[string]string) {
+	tb.Helper()
+
+	root := tb.TempDir()
+	dir := filepath.Join(root, "dir")
+
+	if err := os.Mkdir(dir, 0777); err != nil {
+		tb.Fatalf("Failed to create directory %q: %v", dir, err)
+	}
+	for p, content := range contents {
+		path := filepath.Join(dir, p)
+		parent := filepath.Dir(path)
+		if err := os.Mkdir(parent, 0777); err != nil && !errors.Is(err, fs.ErrExist) {
+			tb.Fatalf("Failed to create directory %q for %q: %v", parent, path, err)
+		}
+		if err := os.WriteFile(path, []byte(content), 0666); err != nil {
+			tb.Fatalf("Failed to write file %q: %v", path, err)
+		}
+	}
+	if err := os.RemoveAll(dst); err != nil && !errors.Is(err, fs.ErrNotExist) {
+		tb.Fatalf("Failed to remove %q: %v", dst, err)
+	}
+	if err := os.Rename(dir, dst); err != nil {
+		tb.Fatalf("Failed to rename from %q to %q: %v", dir, dst, err)
+	}
+}
+
+func TestFileWatcherDir(t *testing.T) {
+	swapSlog(t, slog.New(failSlogHandler{
+		tb:      t,
+		Handler: slog.Default().Handler(),
+	}))
+
+	root := t.TempDir()
+	dir := filepath.Join(root, "dir")
+	if err := os.Mkdir(dir, 0777); err != nil {
+		t.Fatalf("Failed to create directory %q: %v", dir, err)
+	}
+	var parserCalled atomic.Int64
+	parser := filewatcher.WrapDirParser(func(dir fs.FS) (map[string]string, error) {
+		parserCalled.Add(1)
+		m := make(map[string]string)
+		if err := fs.WalkDir(dir, ".", func(path string, de fs.DirEntry, err error) error {
+			if err != nil {
+				return nil
```

Comment:
```suggestion
				return err
```
Unless it's intentionally

Reply:
it is intentional. in the lambda of `WalkDir` when you return non-nil error you stops the walk.